doc/underpinnings.tex

```diff
@@ -0,0 +1,779 @@
+
+For the purposes of our simulation we do not want every run to have
+exactly $40\%$ of the population leave but rather have some variance
+in the size.  How these variances in the population propagate through
+the simulation lets us see confidence bounds when we come to
+aggregate the results of many runs of the model.
+
+A first pass at introducing variance is to use the binomial
+distribution to alter the population that leaves per run.
+
+Continuing our example given $t_{s_1,n_1,ay_1} = 100$ and an
+expectation of $40\%$ then a sample of ten values looks like so:
+
+\begin{lstlisting}
+(sample 10 (binomial {:n 100 :p 0.4}))
+(46 42 31 45 43 44 48 41 41 40)
+\end{lstlisting}
+
+We can further refine the introduction of variance by noting that the
+expectation percentage of leavers is stronger or weaker (more or less
+confident in) depending on the amount of data we have.  Intuitively if
+we have lots of observations around the number of leavers then we can
+be much more confident in our expectation value.  If we have very few
+observations then we are much less confident it's the correct value.
+
+The beta distribution can be used to express the variance in our
+expected leaver percentage.  The parameters for the distribution
+$\alpha$, $\beta$ are simply the observed leavers and non-leavers
+respectively.  Drawing from the distribution is equivalent to working
+out the percentage expectation (obviously slightly changed as it's
+purpose is to introduce variance in this value).  For low values of
+observations there will be a wide range in the values of the expected
+leavers and for high values of observations there will be a much
+narrower range of expected values.
+
+If we use the results of the Beta distribution as the probability
+parameter of the Binomial distribution (composition) we form what's
+called a hierarchical model.  This chaining of distributions increases
+the size of variance in our model runs.
+
+Thus we can write:
+
+\begin{equation*}
+l_{s_x,n_y,ay_z} = \text{Binomial}(t_{s_x,n_y,ay_z},
+\text{Beta}(\alpha^l_{s_x,n_y,ay_z}, \beta^l_{s_x,n_y,ay_z}))
+\end{equation*}
+
+where $\alpha^l_{s_x,n_y,ay_z}$ represents the count of leavers over
+all historic calendar years for the entity $({s_x,n_y,ay_z})$.
+Similarly $\beta^l_{s_x,n_y,ay_z}$ represents the count of all non
+leavers for the same entity over all historic calendar years.
+$(t_{s_x,n_y,ay_z}$ represents the populatio of the entity.
+
+The algorithm described here is a high level overview of the
+implementation by the Send code base.
+
+\subsubsection{Joiners}
+
+The calculation for working out an entity's joiners is split into two
+parts: first the expected rate of joiners is calculated per academic
+year, then this additional population is assigned a need and setting.
+
+The sum of the joiners for a single academic year over all historic
+calendar years is first calculated.
+
+We work out the number of non-joiners by taking the joiners away from
```

Comment:
Not clear why non-joiners are relevant here? Do you mean that you work out the rate of joiners, by seeing what proportion they are of total number of pupils in an AY? (and presumably this is historic data so not 'predicted'?)

Reply:
No, it's the total proportion of the predicted pupil count for an AY.  i.e not the send cohort but all pupils in the school system.